internal/testfixtures/datastore.go

```diff
@@ -119,10 +126,23 @@ func StandardDatastoreWithSchema(ds datastore.Datastore, require *require.Assert
 	validating := NewValidatingDatastore(ds)
 
 	allDefs := []*core.NamespaceDefinition{UserNS, FolderNS, DocumentNS}
+	caveats := []*core.CaveatDefinition{{
+		Name:                 "testcaveat",
+		SerializedExpression: []byte{},
+	}}
+
+	_, _ = ds.ReadWriteTx(ctx, func(ctx context.Context, rwt datastore.ReadWriteTransaction) error {
+		return rwt.WriteCaveats(caveats)
+	})
+	//				require.NoError(err)
 
 	newRevision, err := ds.ReadWriteTx(ctx, func(ctx context.Context, rwt datastore.ReadWriteTransaction) error {
 		for _, nsDef := range allDefs {
-			ts, err := namespace.BuildNamespaceTypeSystemWithFallback(nsDef, rwt, allDefs)
+			ts, err := namespace.NewNamespaceTypeSystem(nsDef,
```

Comment:
This looks very similar to what's done in the service controller method `WriteSchema`, but is duplicating a portion of it. Not necessarily something to address here, but I wonder if we should reuse the logic in the controller by refactoring it so it can be called here.

Reply:
Its been considered to have it use a standard WriteSchema here too, but that would do extra checks that aren't strictly necessary